src/gui/properties/speedwidget.cpp

```diff
@@ -44,6 +42,19 @@
 #include "core/preferences.h"
 #include "core/utils/misc.h"
 
+ComboBoxMenuButton::ComboBoxMenuButton(QWidget *parent, QMenu *menu)
+    : QComboBox(parent)
+    , m_menu(menu)
+{
+}
+
+void ComboBoxMenuButton::showPopup()
+{
+    QPoint p = mapToGlobal(QPoint(0, height()));
+    m_menu->exec(p);
```

Comment:
One last thing. Should you call `QCombobox::hidePopup()` here according to the docs?